manifests/fpm/pool.pp

```diff
@@ -121,50 +129,52 @@
 #   '/etc/php5/fpm/pool.d' or '/etc/php-fpm.d'
 #
 define php::fpm::pool (
-  $ensure                                  = 'present',
-  $listen                                  = '127.0.0.1:9000',
-  $listen_backlog                          = '-1',
-  $listen_allowed_clients                  = undef,
-  $listen_owner                            = undef,
-  $listen_group                            = undef,
-  $listen_mode                             = undef,
-  $user                                    = $php::fpm::config::user,
-  $group                                   = $php::fpm::config::group,
-  Optional[String[1]] $apparmor_hat        = undef,
-  $pm                                      = 'dynamic',
-  $pm_max_children                         = '50',
-  $pm_start_servers                        = '5',
-  $pm_min_spare_servers                    = '5',
-  $pm_max_spare_servers                    = '35',
-  $pm_max_requests                         = '0',
-  $pm_process_idle_timeout                 = '10s',
-  $pm_status_path                          = undef,
-  $ping_path                               = undef,
-  $ping_response                           = 'pong',
-  $access_log                              = undef,
-  $access_log_format                       = '"%R - %u %t \"%m %r\" %s"',
-  $request_terminate_timeout               = '0',
-  $request_slowlog_timeout                 = '0',
-  $security_limit_extensions               = undef,
-  $slowlog                                 = "/var/log/php-fpm/${name}-slow.log",
-  $template                                = 'php/fpm/pool.conf.erb',
-  $rlimit_files                            = undef,
-  $rlimit_core                             = undef,
-  $chroot                                  = undef,
-  $chdir                                   = undef,
-  $catch_workers_output                    = 'no',
-  $include                                 = undef,
-  $env                                     = [],
-  $env_value                               = {},
-  $clear_env                               = true,
-  $options                                 = {},
-  $php_value                               = {},
-  $php_flag                                = {},
-  $php_admin_value                         = {},
-  $php_admin_flag                          = {},
-  $php_directives                          = [],
-  $root_group                              = $php::params::root_group,
-  Optional[Stdlib::Absolutepath] $base_dir = undef,
+  $ensure                                       = 'present',
+  $listen                                       = '127.0.0.1:9000',
+  $listen_backlog                               = '-1',
+  $listen_allowed_clients                       = undef,
+  $listen_owner                                 = undef,
+  $listen_group                                 = undef,
+  $listen_mode                                  = undef,
+  Optional[Array[String[1]]] $listen_acl_users  = undef,
```

Comment:
instead of defaulting to Optional, can you please do this:
```suggestion
  Array[String[1]] $listen_acl_users  = [],
```

this is also documented in our style guide at https://voxpupuli.org/docs/reviewing_pr/

Reply:
Than this
```
  # 'php-fpm' expects a comma separated list of user names
  if $listen_acl_users {
    $real_listen_acl_users = join(flatten($listen_acl_users).unique, ",")
  }
```
should become that
```
  # 'php-fpm' expects a comma separated list of user names
  unless $listen_acl_users.empty {
    $real_listen_acl_users = join(flatten($listen_acl_users).unique, ",")
  }
```

Right?

Reply:
That would work, or you can remove the conditional:
```puppet
$real_listen_acl_users = join(flatten($listen_acl_users).unique, ",")
```
Then change the template to
```ruby
if !@real_listen_acl_users.empty?
```

Reply:
@kenyon What would be the prefered way?

Reply:
Not sure it matters really. Either way.

Reply:
I opted for `unless $listen_acl_users.empty { }`